src/components/Todo.tsx

```diff
@@ -0,0 +1,196 @@
+import classNames from 'classnames';
+import React, {
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { remove, updatingData } from '../api/todos';
+import { Todo } from '../types/Todo';
+
+interface Props {
+  completed: boolean,
+  title: string;
+  id: number;
+  setError: (value: string) => void,
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  todos: Todo[],
+  setSelectedTodoId: (value: number) => void,
+  selectedTodoId: number | null,
+}
+
+export const TodoInfo: React.FC<Props> = ({
+  completed,
+  title,
+  id,
+  setTodos,
+  setError,
+  todos,
+  setSelectedTodoId,
+  selectedTodoId,
+}) => {
+  const [isDoubleClick, setIsDoubleClick] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+
+  const newTodoField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (newTodoField.current) {
+      newTodoField.current.focus();
+    }
+  }, [isDoubleClick]);
+
+  const removeTodo = useCallback(async (removeId: number) => {
+    setSelectedTodoId(removeId);
+    try {
+      await remove(removeId);
+
+      setTodos((state: Todo[]) => [...state]
+        .filter(todo => todo.id !== removeId));
+    } catch (errorFromServer) {
+      setError('Unable to delete a todo');
```

Comment:
also, you could create enum for error messages, it's a good practice not to rely on the unknown string